src/crypto/mod.rs

```diff
@@ -1032,15 +1011,14 @@ impl Serialize for COSEKey {
             COSEKeyType::OKP(_) => 3,
             COSEKeyType::EC2(_) => 5,
             COSEKeyType::RSA(_) => 4,
-            COSEKeyType::Symmetric(_) => 3,
         };
         let mut map = serializer.serialize_map(Some(map_len))?;
         match &self.key {
             COSEKeyType::OKP(key) => {
-                map.serialize_entry(&1, &COSEKeyTypeId::OKP)?;
+                map.serialize_entry(&1, &(COSEKeyTypeId::OKP as u8))?;
                 map.serialize_entry(&3, &self.alg)?;
-                map.serialize_entry(&-1, &key.curve)?;
-                map.serialize_entry(&-2, &key.x)?;
+                map.serialize_entry(&-1, &(key.curve as u8))?;
```

Comment:
I suppose we'll never have more than 256 curves? Actually, does this need to be a u8? Integers don't really have a width in cbor, right? (same with the rest of these u8 casts)

Reply:
The `as u8` is for enum to integer conversion. CBOR integers do have a width, but the serializer chooses the width by value and not type. We could do `as u64` here if we're worried about the range of values.

Reply:
I ended up using i64 instead of u64 because of the presence of negative values in the [registry](https://www.iana.org/assignments/cose/cose.xhtml) for some of these types.

Reply:
LGTM.